staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -158,15 +158,24 @@ type ObjectMeta struct {
 	// +optional
 	UID types.UID `json:"uid,omitempty" protobuf:"bytes,5,opt,name=uid,casttype=k8s.io/kubernetes/pkg/types.UID"`
 
-	// An opaque value that represents the internal version of this object that can
+	// A value that represents the internal version of this object that can
 	// be used by clients to determine when objects have changed. May be used for optimistic
 	// concurrency, change detection, and the watch operation on a resource or set of resources.
-	// Clients must treat these values as opaque and passed unmodified back to the server.
-	// They may only be valid for a particular resource or set of resources.
+	//
+	// This value is a string containing a base-10 encoding of a 63 bit positive integer.
+	// Clients may compare this value for equality between objects. If parsed as above, clients
+	// may compare for recency between two objects of the same API Group and Kind.
```

Comment:
do we have a test that assert we keep this promise 😄 

I think @stevekuznetsov was doing some tests related to this https://github.com/kubernetes/kubernetes/pull/109730, don't know if those can be adapted/reused

Reply:
I think the only current break we know of is events with expiring TTL.

Reply:
In particular, we're (unfortunately) relying on that in watchcache, so it's heavily battle-tested.

Yes - the only case I'm aware when this breaks is https://github.com/kubernetes/kubernetes/issues/110210